### PR TITLE
feat(form-field): Make form field labels trigger input ripples.

### PR DESCRIPTION
**Pull request by sgomes**
This PR creates MDCFormField, which manages the runtime relationship
between an input element and its label, enabling label events to
trigger input ripples. Implements #9.

### CODE REVIEW

File: packages/mdc-form-field/README.md [hunk 1]
```diff
@@ -71,3 +72,81 @@ checkbox:
...
+
+The adapter for `MDCFormField` is extremely simple, providing only methods for adding and
+removing event listeners from the label, as well as methods for activating and deactivating
+the input ripple.
+
+| Method Signature | Description |
+| --- | --- |
+| `registerInteractionHandler(type: string, handler: EventListener) => void` | Adds an event listener `handler` for event type `type` to the label. |
+| `deregisterInteractionHandler(type: string, handler: EventListener) => void` | Removes an event listener `handler` for event type `type` to the label. |
+| `activateInputRipple() => void` | Activates the ripple on the input element. |
```
Comment (traviskaufman) [inline packages/mdc-form-field/README.md]: nit: should we include a link to how to do this with our ripple foundation/adapter here?

File: test/unit/mdc-form-field/foundation.test.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+test('exports strings', (t) => {
+  t.true('strings' in MDCFormFieldFoundation);
+  t.end();
+});
+
+test('defaultAdapter returns a complete adapter implementation', (t) => {
+  const {defaultAdapter} = MDCFormFieldFoundation;
+  const methods = Object.keys(defaultAdapter).filter((k) => typeof defaultAdapter[k] === 'function');
+
+  t.equal(methods.length, Object.keys(defaultAdapter).length, 'Every adapter key must be a function');
```
Comment (traviskaufman) [inline test/unit/mdc-form-field/foundation.test.js]: you can use `verifyDefaultAdapter` within `test/helpers/foundation.js` for this

File: test/unit/mdc-form-field/foundation.test.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  t.end();
+});
+
+function setupTest() {
+  const mockAdapter = td.object(MDCFormFieldFoundation.defaultAdapter);
+  const foundation = new MDCFormFieldFoundation(mockAdapter);
+  return {foundation, mockAdapter};
+}
+
+test('#init calls component and drawer event registrations', (t) => {
```
Comment (traviskaufman) [inline test/unit/mdc-form-field/foundation.test.js]: component and drawer?

**Comment by traviskaufman** [comment]
note: when you squash and merge commit should be `feat(form-field)`, not `feat(mdc-form-field)`

File: test/unit/mdc-form-field/foundation.test.js (hunk 2 above)
Comment (sgomes) [inline test/unit/mdc-form-field/foundation.test.js]: Done.

File: test/unit/mdc-form-field/foundation.test.js (hunk 3 above)
Comment (sgomes) [inline test/unit/mdc-form-field/foundation.test.js]: Bad copy-paste :) Fixed.

File: packages/mdc-form-field/README.md (hunk 1 above)
Comment (sgomes) [inline packages/mdc-form-field/README.md]: Added the clarification.

**Comment by sgomes** [comment]
Changes made, waiting for CI before merging. Thanks, @traviskaufman!

**Pull request merged**